amazon-eks-al2.pkr.hcl

```diff
@@ -0,0 +1,113 @@
+locals {
+  timestamp = regex_replace(timestamp(), "[- TZ:]", "")
+
+  target_ami_name = "${var.ami_name_prefix}-${var.eks_version}-${local.timestamp}"
+
+  e2_test_subnet_id = data.amazon-parameterstore.subnet_id.value
+}
+
+data "amazon-ami" "this" {
+  filters = {
+    architecture        = var.source_ami_arch
+    name                = "${var.ami_name_prefix}-${var.eks_version}-*"
+    root-device-type    = "ebs"
+    state               = "available"
+    virtualization-type = "hvm"
+  }
+
+  most_recent = true
+  owners = [
+    var.source_ami_owner,
+    var.source_ami_owner_govcloud,
+  ]
+  region = var.aws_region
+}
+
+data "amazon-parameterstore" "subnet_id" {
+  name            = "/amazon-eks-custom-amis/tests/subnet-id"
+  with_decryption = false
+  region          = var.aws_region
+}
+
+source "amazon-ebs" "this" {
+  ami_block_device_mappings {
+    delete_on_termination = true
+    device_name           = "/dev/sdb"
+    volume_size           = var.data_volume_size
+    volume_type           = "gp2"
+  }
+
+  ami_description         = "EKS Kubernetes Worker AMI with AmazonLinux2 image"
+  ami_name                = local.target_ami_name
+  ami_virtualization_type = "hvm"
+  instance_type           = var.instance_type
+
+  launch_block_device_mappings {
+    delete_on_termination = true
+    device_name           = "/dev/sda1"
+    volume_size           = var.root_volume_size
+    volume_type           = "gp2"
+  }
+
+  launch_block_device_mappings {
+    delete_on_termination = true
+    device_name           = "/dev/sdb"
+    volume_size           = var.data_volume_size
+    volume_type           = "gp2"
+  }
+
+  region = var.aws_region
+
+  run_tags = {
+    Name = local.target_ami_name
+  }
+
+  source_ami   = data.amazon-ami.this.id
+  ssh_pty      = true
+  ssh_username = var.source_ami_ssh_user
+  subnet_id    = coalesce(var.subnet_id, local.e2_test_subnet_id)
```

Comment:
here we try to use the `subnet_id` provided by the user, or the data source variable from the `tests/vpc` created SSM parameter for e2e testing. (aka - try to use what the user provides, but if nothing is provided lookup from the data source